### PR TITLE
update inventory staff gui

### PR DESCRIPTION
**Pull request by ShaojieDong503**

### CODE REVIEW

File: Android/app/src/main/java/com/example/androidgui/CheckInventoryActivity.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        setContentView(R.layout.activity_check_inventory);
+        button = (Button)findViewById(R.id.button);
+        name = (EditText)findViewById(R.id.textInputEditText);
+        button.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        String iname = name.getText().toString();
+        String info = im.getInfo(iname);
```
Comment (evelyn323) [inline Android/app/src/main/java/com/example/androidgui/CheckInventoryActivity.java:30]: Should have the controller access use case, which then passes information to the presenter (this class) through an output boundary

**Review by ddxxdd-code — CHANGES_REQUESTED** [review]

File: Android/app/src/main/java/com/example/androidgui/UpdateQuanActivity.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        btn = findViewById(R.id.button9);
+        name = findViewById(R.id.textInputEditText9);
+        usage = findViewById(R.id.textInputEditText10);
+        btn.setOnClickListener(this);
+    }
+    @Override
+    public void onClick(View v) {
+        String iname = name.getText().toString();
+        String iusage = usage.getText().toString();
+        String message = im.newQuantity(iname,iusage);
```
Comment (evelyn323) [inline Android/app/src/main/java/com/example/androidgui/UpdateQuanActivity.java:30]: The controller should not directly be returning output to the presenter. Provide information through the output boundary.

(e.g. 1. Presenter calls on method in controller
2. controller calls on method in use case
3. use case calls the output boundary to run a method in the presenter and set the value in the presenter)

File: Android/app/src/main/java/com/example/androidgui/UpdateQuanActivity.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.example.androidgui;
```
[the remark this answers is not in the document]
Comment (ShaojieDong503) [reply Android/app/src/main/java/com/example/androidgui/UpdateQuanActivity.java:1]: Quantity

File: Android/app/src/main/java/controller/inventorySystem/InventoryManager.java [hunk 4]
```diff
@@ -42,9 +40,10 @@ public void newFreshness(String name, String newFreshness){
         }
     }
 
-    public void newQuantity(String name, String usage){
+    public String newQuantity(String name, String usage){
```
Comment (evelyn323) [inline Android/app/src/main/java/controller/inventorySystem/InventoryManager.java:43]: needs javadoc

File: Android/app/src/main/java/controller/inventorySystem/InventoryManager.java [hunk 5]
```diff
@@ -42,9 +40,10 @@ public void newFreshness(String name, String newFreshness){
...
         double u = Double.parseDouble(usage);
-        InventoryList.setQuantity(name, u);
+        String message = InventoryList.setQuantity(name, u);
+        return message;
```
Comment (evelyn323) [inline Android/app/src/main/java/controller/inventorySystem/InventoryManager.java:46]: see previous comment about output boundary

File: Android/app/src/main/java/use_case/kitchen/InventoryList.java [hunk 6]
```diff
@@ -144,12 +144,14 @@ public static double getTotalQuantity(String name){
...
+    public static String setQuantity(String name, double usage) {
+        String message;
         if (!myDict.containsKey(name)){
             //TODO: implement exceptions for cases of wrong key
-            return;
+            message = "wrong name";
         }
-        getItem(name).updateQuantity(usage);
+        message = getItem(name).updateQuantity(usage);
+        return message;
```
Comment (evelyn323) [inline Android/app/src/main/java/use_case/kitchen/InventoryList.java:154]: see above comment about output boundary. Use case should call output boundary to change value in presenter

**Review by evelyn323 — CHANGES_REQUESTED** [review]

**Pull request merged**